### PR TITLE
fix build error for rocal opencl backend

### PR DESCRIPTION
**Pull request by rrawther**
trying to fix build error on OPENCL backend

### CODE REVIEW

File: rocAL/rocAL/include/master_graph.h [hunk 1]
```diff
@@ -110,7 +110,9 @@ class MasterGraph
     void set_sequence_batch_ratio() { _sequence_batch_ratio = _sequence_batch_size / _internal_batch_size; }
     Status get_bbox_encoded_buffers(float **boxes_buf_ptr, int **labels_buf_ptr, size_t num_encoded_boxes);
     size_t bounding_box_batch_count(int* buf, pMetaDataBatch meta_data_batch);
-
+#if ENABLE_OPENCL
+    cl_command_queue get_ocl_cmd_q() {return _device.resources().cmd_queue}
```
Comment (LakshmiKumar23) [inline rocAL/rocAL/include/master_graph.h:114]: @rrawther missing ";"

File: rocAL/rocAL/source/rali_api_meta_data.cpp [hunk 2]
```diff
@@ -235,9 +235,14 @@ RALI_API_CALL raliGetImageLabels(RaliContext p_context, void* buf)
         memcpy(buf, meta_data.second->get_label_batch().data(), sizeof(int) * meta_data_batch_size);
     else
     {
+#if ENABLE_HIP
         hipError_t err = hipMemcpy(buf, meta_data.second->get_label_batch().data(), sizeof(int) * meta_data_batch_size, hipMemcpyHostToDevice);
         if (err != hipSuccess)
             THROW("Invalid Data Pointer: Error copying to device memory")
+#else
+        if(clEnqueueWriteBuffer(context->master_graph->get_ocl_cmd_q(), buf, CL_TRUE, 0, sizeof(int) * meta_data_batch_size, meta_data.second->get_label_batch().data(), 0, NULL, NULL) != CL_SUCCESS)
```
Comment (LakshmiKumar23) [inline rocAL/rocAL/source/rali_api_meta_data.cpp:243]: @rrawther  error: invalid conversion from 'void*' to 'cl_mem' {aka '_cl_mem*'}

File: rocAL/rocAL/include/master_graph.h (hunk 1 above)
Comment (rrawther) [reply rocAL/rocAL/include/master_graph.h:114]: Sorry because I couldn't build it

File: rocAL/rocAL/source/rali_api_meta_data.cpp [hunk 3]
```diff
@@ -313,14 +318,19 @@ RALI_API_CALL raliGetOneHotImageLabels(RaliContext p_context, void* buf, int num
...
-        memcpy(buf,one_hot_encoded, sizeof(int) * meta_data_batch_size * numOfClasses);
+        memcpy(buf, one_hot_encoded, sizeof(int) * meta_data_batch_size * numOfClasses);
     else
     {
+#if ENABLE_HIP
         hipError_t err = hipMemcpy(buf, one_hot_encoded, sizeof(int) * meta_data_batch_size * numOfClasses, hipMemcpyHostToDevice);
         if (err != hipSuccess)
             THROW("Invalid Data Pointer: Error copying to device memory")
+#else
+        if(clEnqueueWriteBuffer(context->master_graph->get_ocl_cmd_q(), buf, CL_TRUE, 0, sizeof(int) * meta_data_batch_size * numOfClasses, one_hot_encoded, 0, NULL, NULL) != CL_SUCCESS)
```
Comment (LakshmiKumar23) [inline rocAL/rocAL/source/rali_api_meta_data.cpp:330]: @rrawther error: invalid conversion from 'void*' to 'cl_mem' {aka '_cl_mem*'}

File: rocAL/rocAL/include/master_graph.h (hunk 1 above)
Comment (LakshmiKumar23) [reply rocAL/rocAL/include/master_graph.h:114]: no problem :)

**Pull request merged**